src/checkers/inference/util/SlotDefaultTypeResolver.java

```diff
@@ -185,5 +187,16 @@ public Void visitAnnotatedType(AnnotatedTypeTree tree, Void unused) {
 
             return super.visitAnnotatedType(tree, unused);
         }
+
+        @Override
+        public Void visitNewClass(NewClassTree tree, Void unused) {
+            AnnotatedTypeMirror defaultType = getDefaultTypeFor(tree);
+            ExpressionTree type = tree.getIdentifier();
+            if ((type instanceof ParameterizedTypeTree) && !((ParameterizedTypeTree) type).getTypeArguments().isEmpty())
+                return super.visitNewClass(tree, unused);
```

Comment:
We probably don't need the logic here. We should always save `defaultType` to the cache before visiting child nodes.

Reply:
Without the `if` branch, CI failed on the test case `AnonymousProblem.java`. It raised the assertion error from the `visitParameterizedType`
The reason is because: (take `new ArrayList<String>() as an example`). the NewClassTree does not have String as the type argument, which has 0 type argument, but the ParameterizedTypeTree has 1 type argument, and thus cause the assertion failure.